presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -589,7 +595,11 @@ private S3Object getS3Object(final Path path, final long start)
                                 return s3.getObject(new GetObjectRequest(host, keyFromPath(path)).withRange(start, Long.MAX_VALUE));
                             }
                             catch (AmazonServiceException e) {
-                                if (e.getStatusCode() == SC_FORBIDDEN) {
+                                if (e.getStatusCode() == SC_REQUESTED_RANGE_NOT_SATISFIABLE) {
+                                    log.info("The given starting byte(%d) was past the end of file(%s), ignore it. (%s)", start, path, e.getMessage());
+                                    return new S3Object();
```

Comment:
I'm not sure this works, because this object won't have an input stream, and the method `openStream()` assumes that it does. Since `openStream()` is the only caller of this method, I think we should rename this method to `getS3InputStream()` that returns an `S3ObjectInputStream`, and return an empty stream like this:

```java
return new S3ObjectInputStream(new ByteArrayInputStream(new byte[0]), null);